### PR TITLE
Feature/cosmos emul enhancements

### PR DESCRIPTION
**Pull request by kforeverisback**
# Type of PR

- [X] Documentation changes
- [X] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [X] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] My code follows the code style of this project.
- [ ] I ran lint checks locally prior to submission.
- [X] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR

- Modify Cosmos Emulator startup and certificate generation to make sure it works with both local development and k3d environment.
- Added new `loderunner-api` deployment with proper lr-secrets and certificates for Cosmos Emulator
- Generate SSL cert to work with local cosmos url as well as k3d internal-to-host URL (host.k3d.internal)

## Does this introduce a breaking change

- [ ] YES
- [X] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [X] Update documentation or issue referenced above

Follow the [instructions](https://github.com/retaildevcrews/loderunner#running-the-system-via-codespaces) to create k3d cluster in Codespaces using CosmosDB Emulator.
For quick verification, on Step 7 use `make lr-local-emul-api-only`

Summary of steps to follow in Readme:

```bash
# From the REPO root
# Export proper variables to be used by k8s secrets
export LR_DB=LodeRunnerDB
export LR_COL=LodeRunner

# Create k3d cluster
make create

# Create LodeRunner API deployment with Cosmos Emulator
make lr-local-emul-api-only
```
 
## Issues Closed or Referenced

- Closes https://github.com/retaildevcrews/wcnp/issues/647

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -110,6 +111,63 @@ clean :
...
 	@kubectl get po -A
 
+lr-local-emul-api-only:
+	# Create LodeRunner.API image from codebase
+	@#docker build ./src -t k3d-registry.localhost:5000/loderunner-api:local -f ./src/LodeRunner.API/Dockerfile
+	# Load new LodeRunner.API image to k3d registry
+	@#docker push k3d-registry.localhost:5000/loderunner-api:local
+
+	# Delete previous deployed LodeRunner apps
+	-kubectl delete -f deploy/loderunner/local-comos/3-loderunner-api.yaml --ignore-not-found=true
```
Comment (PurpleBriar) [inline Makefile:121]: Update to:

-kubectl delete -f deploy/loderunner/local-cosmos/3-loderunner-api.yaml --ignore-not-found=true

**Comment by PurpleBriar** [comment]
I got this error when I ran the command "make lr-local-emul-api-only"

`# Create NGINX secrets for Cosmos Emulator
error: error reading /workspaces/cosmos-emulator/nginx/cosmos-linux-emulator.crt: no such file or directory`
`make: *** [Makefile:126: lr-local-emul-api-only] Error 1`

**Comment by kforeverisback** [comment]
> I got this error when I ran the command "make lr-local-emul-api-only"
> 
> `# Create NGINX secrets for Cosmos Emulator error: error reading /workspaces/cosmos-emulator/nginx/cosmos-linux-emulator.crt: no such file or directory` `make: *** [Makefile:126: lr-local-emul-api-only] Error 1`

Fixed Certificate install issue. Please try again now.
PS: Also added status logs

File: deploy/loderunner/local-cosmos/3-loderunner-api.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    matchLabels:
+      app: loderunner-api
+  template:
+    metadata:
+      labels:
+        app: loderunner-api
+    spec:
+      containers:
+        - name: app
+          image: k3d-registry.localhost:5000/loderunner-api:local
```
Comment (PurpleBriar) [inline deploy/loderunner/local-cosmos/3-loderunner-api.yaml:20]: I'm having trouble pulling this image in codespaces. The error from the pod is:

`Failed to pull image "k3d-registry.localhost:5000/loderunner-api:local": rpc error: code = NotFound desc = failed to pull and unpack image "k3d-registry.localhost:5000/loderunner-api:local": failed to resolve reference "k3d-registry.localhost:5000/loderunner-api:local": k3d-registry.localhost:5000/loderunner-api:local: not found`

File: Makefile [hunk 3]
```diff
@@ -110,6 +111,63 @@ clean :
 	# show running pods
 	@kubectl get po -A
 
+lr-local-emul-api-only:
+	# Create LodeRunner.API image from codebase
+	@#docker build ./src -t k3d-registry.localhost:5000/loderunner-api:local -f ./src/LodeRunner.API/Dockerfile
```
Comment (PurpleBriar) [inline Makefile:116]:
```suggestion
	@docker build ./src -t k3d-registry.localhost:5000/loderunner-api:local -f ./src/LodeRunner.API/Dockerfile
```

File: Makefile [hunk 4]
```diff
@@ -110,6 +111,63 @@ clean :
...
+	# Load new LodeRunner.API image to k3d registry
+	@#docker push k3d-registry.localhost:5000/loderunner-api:local
```
Comment (PurpleBriar) [inline Makefile:118]:
```suggestion
	@docker push k3d-registry.localhost:5000/loderunner-api:local
```

**Review by PurpleBriar — APPROVED** [review]

**Comment by kforeverisback** [comment]
Fixed a bug where sometimes, after resuming a Codespace, Cosmos emulator certificate was invalid (deleted). Fixed it by `sleep 5` then rechecking Cosmos Cert and downloading it again

**Pull request merged**